### DISCUSSION TITLE
Document the Python module?

### DISCUSSION THREAD

**Discussion by Mineinjava** [LennyPhoenix/py-discord-sdk#5] [General]
## Currently the only documentation for this repo is the discord developer docs
It might be nice to add documentation for the python module too, using either the [wiki](https://github.com/LennyPhoenix/py-discord-sdk/wiki) or [readTheDocs](https://readthedocs.org). 

I would be willing to help document this project.

**Comment by cmontigel** [comment]
Already on the to do list.
See here:
![image](https://user-images.githubusercontent.com/67985645/106822424-fa155780-6633-11eb-8189-bd2110ea17e1.png)
(from readme.md)

**Comment by LennyPhoenix** [comment]
If I'm honest, I haven't been maintaining this much recently but I would be willing to sort out some documentation. I can start work on it tomorrow :+1:

**Reply by Mineinjava** [reply]
great! I will attempt to add some more doctstrings once I mess with the module a bit more

**Reply by Mineinjava** [reply]
we should also keep in mind specifying what does and doesn't need a dev license

**Reply by LennyPhoenix** [reply]
yeah that would probably be a good idea

**Comment by LennyPhoenix** [comment]
Currently working on the docs now, you can take a look at the wiki for progress.

**Comment by Mineinjava** [comment]
sorry I haven't been able to work on this much. You may consider using sphinx-autodoc for the documentation

**Reply by Mineinjava** [reply]
I *could* set this up via PR, but I want to reach a consensus first.

**Reply by LennyPhoenix** [reply]
That sounds like a pretty good idea, and it would probably be more maintainable than manual documentation in [the wiki](https://github.com/LennyPhoenix/py-discord-sdk/wiki), feel free to open up a PR for it if you want to set it up. I can work on adding all the docstrings to the classes and functions at some point (I believe quite a few are already documented), although I haven't worked on the module much recently.

**Reply by Mineinjava** [reply]
made the PR (#6). you still need to add it to ReadTheDocs.

**Reply by Mineinjava** [reply]
we should also use the [sphinx format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html "RST Based Docstrings") for docstrings.
